UploadDaemon/UploadTask.cs

```diff
@@ -128,7 +128,8 @@ private void ProcessTraceFile(TraceFile trace, Archive archive, Config config, L
                 return;
             }
 
-            Config.ConfigForProcess processConfig = config.CreateConfigForProcess(processPath);
+            ParsedTraceFile parsedTraceFile = new ParsedTraceFile(trace.Lines, trace.FilePath);
+            Config.ConfigForProcess processConfig = config.CreateConfigForProcess(processPath, parsedTraceFile);
```

Comment:
Currently I get a lot of errors if no `loadedAssemblyPathRegex` matches on my traces (got 7 error messages for one "ProfilerTestee" upload), maybe we could be a bit more compact with the error handling? Here is the stacktrace:

```
2023-04-04 22:31:15.6695|ERROR|UploadDaemon.UploadTask|Failed to process trace file "D:\teamscale-profiler-dotnet\Profiler\bin\Debug\coverage_20230404_2030590954.txt". Will retry later|InvalidConfigException Config for process D:\fedex-2023-03-29-dotner-profiler-installer\source\ProfilerTestee\bin\x86\De bug\ProfilerTestee.exe is invalid: Invalid configuration for process D:\fedex-2023-03-29-dotner-profiler-installer\source\ProfilerTestee\bin\x86\De bug\ProfilerTestee.exe. You must provide either a Teamscale server (property "teamscale") or a directory (property "directory") or an Azure File Storage (property "azureFileStorage") or an Artifactory (property "artifactory") to upload coverage files to., Invalid configuration for process D:\fedex-2023-03-29-dotner-profiler-installer\source\ProfilerTestee\bin\x86\De bug\ProfilerTestee.exe. You must provide an assembly name (property "versionAssembly", without the file extension) to read the program version from in order to upload method coverage. Alternatively, you can configure line coverage upload (properties "pdbDirectory" and "revisionFile").    at UploadDaemon.Configuration.Config.CreateConfigForProcess(String profiledProcessPath, ParsedTraceFile traceFile) in D:\teamscale-profiler-dotnet\UploadDaemon\Configuration\Config.cs:line 274
   at UploadDaemon.UploadTask.ProcessTraceFile(TraceFile trace, Archive archive, Config config, LineCoverageMerger coverageMerger) in D:\teamscale-profiler-dotnet\UploadDaemon\UploadTask.cs:line 132
   at UploadDaemon.UploadTask.ScanDirectory(String traceDirectory, Config config) in D:\teamscale-profiler-dotnet\UploadDaemon\UploadTask.cs:line 58
```